### ISSUE TITLE
LogisticCalibration implementation differences to scikit-learn

### ISSUE THREAD

**Issue by deutschmn** [fabiankueppers/calibration-framework#28]
Hi @fabiankueppers,

Thanks for creating this great library. It works perfectly for our use case 😊 There's just one thing I'm wondering about:

For `LogisticCalibration`, the documentation states that it implements Platt scaling. However, I've found that it yields quite different results than when implementing it with the [logistic model in sklearn](https://scikit-learn.org/stable/modules/calibration.html#sigmoid).

So this

```python
from netcal.scaling import LogisticCalibration

LC = LogisticCalibration()
LC.fit(np.array(pred), np.array(labels))
calibrated_prob = LC.transform(np.array(pred))
```

gives very different results from this:

```python
from sklearn.linear_model import LogisticRegression as LR
lr = LR().fit(np.reshape(pred,(-1,1)), labels)
calibrated_prob = lr.predict_proba(np.reshape(pred,(-1,1)))[:,1]
```

Are there any intended differences between your implementation and sklearn? Or are we just comparing it wrong?

I've found it challenging to tell by looking at the code alone. 

Thanks,
Patrick

**Comment by fabiankueppers** [comment]
Hi @deutschmn,
the `LogisticCalibration` tool does a little bit more than just applying logistic regression. According to [Guo et al.](https://arxiv.org/pdf/1706.04599.pdf), the rescaling is not directly applied to the probabilities but rather to the logits, e.g., of a neural network. Since the input to `LogisticCalibration` is asserted to consist of the probabilities, the method applies an inverse sigmoid/softmax (depending if you provide binary or multiclass classification) before logistic regression. This might explain the differences. 
Please take a look at lines 182
https://github.com/fabiankueppers/calibration-framework/blob/f1995512fea511572171974913fe5b569cac0cd0/netcal/scaling/LogisticCalibration.py#L182
and https://github.com/fabiankueppers/calibration-framework/blob/f1995512fea511572171974913fe5b569cac0cd0/netcal/scaling/LogisticCalibration.py#L186
of [LogisticCalibration.py](https://github.com/fabiankueppers/calibration-framework/blob/master/netcal/scaling/LogisticCalibration.py).

Hope this helps. If so, please close the issue. Thank you!
Fabian
